<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Spyking circus launcher</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="lbl_file">
         <property name="text">
          <string>Data file</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="edit_file">
         <property name="toolTip">
          <string>The data file (or file with list of commands in batch mode)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btn_file">
         <property name="text">
          <string>Browse...</string>
         </property>
         <property name="icon">
          <iconset theme="system-file-manager">
           <normaloff/>
          </iconset>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QFrame" name="frame">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QGroupBox" name="grp_tasks">
            <property name="title">
             <string>Tasks</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_3">
               <item>
                <widget class="QCheckBox" name="cb_filtering">
                 <property name="text">
                  <string>Filtering</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="cb_whitening">
                 <property name="text">
                  <string>Whitening</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="cb_clustering">
                 <property name="text">
                  <string>Clustering</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="cb_fitting">
                 <property name="text">
                  <string>Fitting</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="cb_merging">
                 <property name="text">
                  <string>Merging</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="cb_converting">
                 <property name="text">
                  <string>Converting</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="cb_gathering">
                 <property name="text">
                  <string>Gathering</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="cb_extracting">
                 <property name="text">
                  <string>Extracting</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="cb_benchmarking">
                 <property name="text">
                  <string>Benchmarking</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="Line" name="line">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="cb_results">
                 <property name="text">
                  <string>Results</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="grp_options">
            <property name="title">
             <string>Options</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_6">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_10">
               <item>
                <widget class="QCheckBox" name="cb_batch">
                 <property name="toolTip">
                  <string>Run a set of commands from a file</string>
                 </property>
                 <property name="text">
                  <string>Batch mode</string>
                 </property>
                 <property name="autoExclusive">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="cb_preview">
                 <property name="toolTip">
                  <string>Open a GUI showing the first second filtered with thresholds</string>
                 </property>
                 <property name="text">
                  <string>Preview mode</string>
                 </property>
                 <property name="autoExclusive">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="Line" name="line_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_11">
               <item>
                <widget class="QLabel" name="lbl_cpus">
                 <property name="text">
                  <string>No. CPUs</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spin_cpus">
                 <property name="toolTip">
                  <string>Number of CPUs to use</string>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="lbl_gpus">
                 <property name="text">
                  <string>No. GPUs</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spin_gpus">
                 <property name="toolTip">
                  <string>Number of GPUs to use</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_7">
               <item>
                <widget class="QLabel" name="lbl_hostfile">
                 <property name="text">
                  <string>Host file</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="edit_hostfile">
                 <property name="toolTip">
                  <string>The hostfile for MPI</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="btn_hostfile">
                 <property name="text">
                  <string>Browse...</string>
                 </property>
                 <property name="icon">
                  <iconset theme="system-file-manager">
                   <normaloff/>
                  </iconset>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QGroupBox" name="grp_merge_convert">
               <property name="title">
                <string>Merging/Converting</string>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_7">
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_6">
                  <item>
                   <widget class="QLabel" name="lbl_extension">
                    <property name="text">
                     <string>Extension</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLineEdit" name="edit_extension"/>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="grp_benchmarking">
               <property name="title">
                <string>Benchmarking</string>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_5">
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_8">
                  <item>
                   <widget class="QLabel" name="lbl_output">
                    <property name="text">
                     <string>Output</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QLineEdit" name="edit_output">
                    <property name="toolTip">
                     <string>Output file for generation of synthetic benchmarks</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QPushButton" name="btn_output">
                    <property name="text">
                     <string>Browse...</string>
                    </property>
                    <property name="icon">
                     <iconset theme="system-file-manager">
                      <normaloff/>
                     </iconset>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_9">
                  <item>
                   <widget class="QLabel" name="lbl_type">
                    <property name="text">
                     <string>Type</string>
                    </property>
                    <property name="alignment">
                     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QComboBox" name="cmb_type">
                    <property name="toolTip">
                     <string>type of benchmark</string>
                    </property>
                    <item>
                     <property name="text">
                      <string>fitting</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>clustering</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>synchrony</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QPlainTextEdit" name="edit_command">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>40</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>Monospace</family>
        </font>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="frame_2">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <property name="rightMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="btn_run">
          <property name="toolTip">
           <string>Run spyking circus with the given options</string>
          </property>
          <property name="text">
           <string>Run</string>
          </property>
          <property name="icon">
           <iconset theme="system-run">
            <normaloff/>
           </iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_stop">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Interrupt a running spyking-circus process</string>
          </property>
          <property name="text">
           <string>Stop</string>
          </property>
          <property name="icon">
           <iconset theme="process-stop">
            <normaloff/>
           </iconset>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>495</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="btn_close">
          <property name="toolTip">
           <string>Close the window</string>
          </property>
          <property name="text">
           <string>Close</string>
          </property>
          <property name="icon">
           <iconset theme="window-close">
            <normaloff/>
           </iconset>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QTextEdit" name="edit_stdout">
       <property name="font">
        <font>
         <family>Monospace</family>
        </font>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>btn_close</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>489</x>
     <y>396</y>
    </hint>
    <hint type="destinationlabel">
     <x>316</x>
     <y>354</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
